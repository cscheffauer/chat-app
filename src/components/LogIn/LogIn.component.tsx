import React, { FormEvent, useState } from 'react'
import './LogIn.scss';

interface Props {
    logInUser: (event: any) => void
}

const LogIn = ({ logInUser }: Props) => {
    const [username, setusername] = useState("");

    const changeUsername = (event: FormEvent<HTMLInputElement>) => {
        event.preventDefault();
        setusername(event.currentTarget.value);
    }

    const handleSubmit = (event: FormEvent) => {
        event.preventDefault();
        logInUser(username)
    }
    return (
        <div className="login">
            <p>Please type in your name to join:</p>
            <form onSubmit={handleSubmit}>
                <input onChange={changeUsername}></input>
                <button type="submit">Join</button>
            </form>
        </div>
    )
}

export default LogIn;
