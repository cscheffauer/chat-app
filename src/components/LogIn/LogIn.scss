.login {
	text-align: center;
	margin-top: 5vh; /* Fallback for browsers that do not support Custom Properties */
	margin-top: calc(var(--vh, 1vh) * 5);
}
