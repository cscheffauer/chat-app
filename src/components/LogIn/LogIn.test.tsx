import React from 'react';
import { shallow } from 'enzyme';
import LogIn from './LogIn.component';


describe('Component / LogIn', () => {
    const logInUserMock = () => { };
    const LogInWrapper = shallow(<LogIn logInUser={logInUserMock} />);

    it('expect to render LogIn component', () => {
        expect(LogInWrapper).toMatchSnapshot();
    })

})


