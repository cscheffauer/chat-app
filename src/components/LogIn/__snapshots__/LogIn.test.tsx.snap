// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component / LogIn expect to render LogIn component 1`] = `ShallowWrapper {}`;
